projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -355,3 +356,29 @@ export function resizeObservable(target: HTMLElement): Observable<ResizeObserver
         return unsubscribe;
     });
 }
+
+/**
+ * @hidden
+ * @internal
+ *
+ * Compares two metadata maps.
+ */
+export function compareMetadata(map1: Map<any, any>, map2: Map<any, any>) {
+    let metadataMatch = true;
+    if (map1 && map2) {
```

Comment:
If only one of the maps is falsy, the method will return true, which is not correct.